ecs-cli/modules/cli/local/down_app.go

```diff
@@ -0,0 +1,115 @@
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package local
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/network"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+)
+
+// TODO These labels should be defined part of the local.Create workflow.
+// Refactor to import these constants instead of re-defining them here.
+const (
+	ecsLocalDockerComposeFileName = "docker-compose.local.yml"
+)
+
+// Down stops and removes a running local ECS task.
+// If the user stops the last running task in the local network then also remove the network.
+func Down(c *cli.Context) error {
+	defer func() {
+		docker := newDockerClient()
+		network.Teardown(docker)
+	}()
+
+	taskPath := c.String(flags.TaskDefinitionFileFlag)
+	taskARN := c.String(flags.TaskDefinitionArnFlag)
+
+	// TaskDefinitionFileFlag flag has priority over TaskDefinitionArnFlag if both are present
+	if taskPath != "" {
+		return handleDownWithFile(taskPath)
+	}
+	if taskARN != "" {
+		return handleDownWithARN(taskARN)
+	}
+	return handleDownWithCompose()
+}
+
+func handleDownWithFile(path string) error {
+	return handleDownWithFilters(filters.NewArgs(
+		filters.Arg("label", fmt.Sprintf("%s=%s", taskFilePathLabelKey, path)),
+	))
+}
+
+func handleDownWithARN(value string) error {
+	return handleDownWithFilters(filters.NewArgs(
+		filters.Arg("label", fmt.Sprintf("%s=%s", taskDefinitionARNLabelKey, value)),
+	))
+}
+
+func handleDownWithCompose() error {
+	wd, _ := os.Getwd()
+	if _, err := os.Stat(filepath.Join(wd, ecsLocalDockerComposeFileName)); os.IsNotExist(err) {
+		logrus.Warnf("Compose file %s does not exist in current directory", ecsLocalDockerComposeFileName)
+		return nil
+	}
+
+	logrus.Infof("Running Compose down on %s", ecsLocalDockerComposeFileName)
+	cmd := exec.Command("docker-compose", "-f", ecsLocalDockerComposeFileName, "down")
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		logrus.Fatalf("Failed to run docker-compose down due to %v", err)
+	}
+
+	logrus.Info("Stopped and removed containers successfully")
+	return nil
+}
+
+func handleDownWithFilters(args filters.Args) error {
+	docker := newDockerClient()
+
+	containers, err := docker.ContainerList(context.Background(), types.ContainerListOptions{
+		Filters: args,
+		All:     true,
+	})
+	if err != nil {
+		logrus.Fatalf("Failed to list containers with args %v due to %v", args, err)
+	}
+	if len(containers) == 0 {
+		logrus.Warnf("No containers found with label %v", args.Get("label"))
+		return nil
+	}
+
+	for _, container := range containers {
+		if err = docker.ContainerStop(context.Background(), container.ID, nil); err != nil {
```

Comment:
Can we use a context with a timeout

Reply:
Yes! 🙈 done, moved docker.go to its own sub-package with an exported timeout constant.